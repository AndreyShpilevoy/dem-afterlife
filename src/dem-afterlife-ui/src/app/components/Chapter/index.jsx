import React from 'react';
import {sharedPropTypes, defaults} from 'utils';
import CollapsibleSection from 'components/CollapsibleSection';
import Forum from 'components/Forum';
import Term from 'containers/Term';

const topicsTerm = {id: 1, value: 'Topics'};
const postsTerm = {id: 2, value: 'Posts'};
const lastMessageTerm = {id: 3, value: 'Last message in'};

const headerSettings = title => ({
    title,
    firstColumnTerm: <Term term={topicsTerm} />,
    secondColumnTerm: <Term term={postsTerm} />,
    thirdColumnTerm: <Term term={lastMessageTerm} />
});

const collapseSettings = {
    collapsedByDefault: true,
    isCollapsible: true
};

const mapForums = forumArray => forumArray.map(x => <Forum key={x.id} forum={x} />);

const Chapter = ({chapter}) =>
    (<CollapsibleSection headerSettings={headerSettings(chapter.title)} collapseSettings={collapseSettings}>
        {chapter.forumArray ? mapForums(chapter.forumArray) : defaults.emptyString}
    </CollapsibleSection>);


Chapter.propTypes = {
    chapter: sharedPropTypes.chapter.isRequired
};

export default Chapter;