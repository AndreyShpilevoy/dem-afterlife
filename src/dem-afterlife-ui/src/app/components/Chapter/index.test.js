/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import Chapter from './index';

jest.mock('components/Forum');

describe('Chapter', () => {
    it('component match expected snapshot without forumArray', () => {
        const chapter = {id: 2, title: 'Ex Machina Меридиан 113', order: 2, forumArray: [] };
        expect(shallow(<Chapter className='chapterClassName' chapter={chapter} />)).toMatchSnapshot();
    });

    it('component match expected snapshot with forumArray', () => {
        const chapter = {
            id: 2,
            title: 'Ex Machina Меридиан 113',
            order: 2,
            forumArray: [{
                id: 40,
                chapterId: 4,
                order: 4,
                title: 'Вопросы по созданию моделей',
                description: 'Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.',
                topicsCount: 4,
                postsCount: 864,
                lastTopicInfo: {
                    topicId: 4,
                    topicDescription: 'Exporter Ex Machina for Maya',
                    latestPostTimeCreation: new Date('2007/09/06 22:33:09'),
                    latestPostAuthorId: 4,
                    latestPostAuthorName: 'Buba',
                    latestPostAuthorGroupColor: '#00AA00'
                } }
            ]
        };
        expect(shallow(<Chapter className='chapterClassName' chapter={chapter} />)).toMatchSnapshot();
    });
});