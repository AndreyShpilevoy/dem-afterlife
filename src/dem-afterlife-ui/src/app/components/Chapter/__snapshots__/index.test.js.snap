// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter component match expected snapshot with forumArray 1`] = `
<Jss(CollapsibleSectionPure)
  collapseSettings={
    Object {
      "collapsedByDefault": true,
      "isCollapsible": true,
    }
  }
  headerSettings={
    Object {
      "firstColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 1,
            "value": "Topics",
          }
        }
      />,
      "secondColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 2,
            "value": "Posts",
          }
        }
      />,
      "thirdColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 3,
            "value": "Last message in",
          }
        }
      />,
      "title": "Ex Machina Меридиан 113",
    }
  }
>
  <Jss(ForumPure)
    forum={
      Object {
        "chapterId": 4,
        "description": "Вопросы по созданию моделей, конвертированию в игру и всего что с этим связано.",
        "id": 40,
        "lastTopicInfo": Object {
          "latestPostAuthorGroupColor": "#00AA00",
          "latestPostAuthorId": 4,
          "latestPostAuthorName": "Buba",
          "latestPostTimeCreation": 2007-09-06T19:33:09.000Z,
          "topicDescription": "Exporter Ex Machina for Maya",
          "topicId": 4,
        },
        "order": 4,
        "postsCount": 864,
        "title": "Вопросы по созданию моделей",
        "topicsCount": 4,
      }
    }
  />
</Jss(CollapsibleSectionPure)>
`;

exports[`Chapter component match expected snapshot without forumArray 1`] = `
<Jss(CollapsibleSectionPure)
  collapseSettings={
    Object {
      "collapsedByDefault": true,
      "isCollapsible": true,
    }
  }
  headerSettings={
    Object {
      "firstColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 1,
            "value": "Topics",
          }
        }
      />,
      "secondColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 2,
            "value": "Posts",
          }
        }
      />,
      "thirdColumnTerm": <Connect(Term)
        term={
          Object {
            "id": 3,
            "value": "Last message in",
          }
        }
      />,
      "title": "Ex Machina Меридиан 113",
    }
  }
/>
`;
